Scripting documention improvements and naming changes (#104)
BeamMP/Docs: docs/en/scripting/server/latest-server-reference.md

If the event with that name doesn't exist, it's created, and thus RegisterEvent cannot fail. This can be used to create custom events.

Example:

```lua
function ChatHandler(player_id, player_name, msg)

>     if msg == "damn" then

Reviewer: could we change `damn` for something else like `hello`?

You can use this, for example, to tell a player *why* you cancelled their vehicle spawn, chat message, or similar, or to display some information about your server.

Example:
```lua
function ChatHandler(player_id, player_name, msg)
    if string.match(msg, "damn") then

>         MP.SendChatMessage(player_id, "Please do not use profanity.")

Reviewer 2: ```suggestion
        MP.SendChatMessage(player_id, "Please do not use profanity.") -- If the message has the text "damn" in it, notify the player and cancel their message.
```
Could we change `damn` to something like `darn`? And would you mind making another example which is just sending a broadcast message (id `-1`)?

if error == "" then
    print(raw_pos)
else
    print(error)
end
```
Output:
```json
 {
    tim: 49.824, // Time since spawn
    rvel: { // Rotational velocity
            1: -1.33564e-05,
            2: -9.16553e-06,
            3: 8.33364e-07,
    }, 
    vel: { // Velocity
            1: -4.29755e-06,
            2: -5.79335e-06,
            3: 4.95236e-06,
    },
    pos: { // Position
            1: 269.979,
            2: -759.068,
            3: 46.554,
    },
    ping: 0.0125, // Vehicle latency
    rot: { // Rotation
            1: -0.00559953,
            2: 0.00894832,
            3: 0.772266,
            4: 0.635212,
    },
}
```
Example 2:
```lua
local player_id = 4
local vehicle_id = 0

local raw_pos, error = MP.GetPositionRaw(player_id, vehicle_id)
if error = "" then
    local x, y, z = table.unpack(raw_pos["pos"])

    print("X:", x)
    print("Y:", y)
    print("Z:", z)
else
    print(error)
end
```
Output:
```
X: -603.459
Y: -175.078
Z: 26.9505
```

#### `MP.IsPlayerConnected(player_id: number) -> boolean`

Whether the player is connected and if the server has received a UDP packet from them.

Example:
```lua
local player_id = 8

> local conn_status = MP.IsPlayerConnected(player_id)
> 
> print(connStatus)

Reviewer 2: ```suggestion
print(MP.IsPlayerConnected(player_id)) -- Check if player with ID 8 is properly connected.
```

Whether the player is a guest. A guest is someone who didn't log in, and instead chose to play as a guest. Their name is usually `guest` followed by a long number.

As guests are anonymous, you may want to disallow them to join, however it is recommended you use the [`onPlayerAuth`](#onplayerauth) `is_guest` argument.

#### `MP.DropPlayer(player_id: number, [reason: string])`

Kicks the player with the specified ID. The reason parameter is optional.

```lua
local player_id = 3
local vcount = 0
local player_vehicles = MP.GetPlayerVehicles(player_id)

for vehicle_id, vehicle_data in pairs(player_vehicles) do
    vcount = vcount + 1
    if vcount > 1 then
        MP.RemoveVehicle(player_id, vehicle_id)
        MP.DropPlayer(player_id, "You can only have one vehicle spawned at once.")
    end

> end

Reviewer 2: Could this be turned into something like getting kicked for sending a certain chat message instead?

Example:

```lua
local player_id = 5

> local ident = MP.GetPlayerIdentifiers(player_id)
> 
> print(ident)

Reviewer 2: ```suggestion
print(MP.GetPlayerIdentifiers(player_id))
```

```
Output:
```json
{
    ip: "127.0.0.1",

>     discord: "680149671821181011",

Reviewer: Just check that you want to actually have your Discord and BeamMP ID in here.

>     discord: "680149671821181011",

Reviewer 2: Please specify that the discord id will only show up if the user has linked it.

>     beammp: "1100385",

Reviewer 2: Could we get something here about how to get your beammp id (go to `https://forum.beammp.com/u/USERNAME.json` and look for `"user": { "id":`)  and how it cannot be changed unlike a username?

Sets a ServerConfig setting temporarily. For this, the `MP.Settings` table is useful.

Example:

> Turning on Debug mode

Reviewer 2: Could this be reverted?
